### ISSUE TITLE
Keystrokes 'hanging around' in echo area after embark-act

### ISSUE THREAD

**Issue by codesuki** [oantolin/embark#470]
Hi! First off, awesome package! 

When I run `C-x C-f `(find-file), select a file, run `embark-act`, and press `RET` (find-file), the echo-area shows `RET-` waiting for a key-chord. If I then for example press `s` it shows `RET-s`, disappears, and inserts `s` into the buffer.

It doesn't really have a negative impact, but I was thinking maybe it hints at some problem (Why is it waiting for a command?). I was looking at the code of `embark--act` and it does manipulate `this-command-keys` which could be connected (I don't know enough about how it works).

**Comment by oantolin** [comment]
I haven't been able to reproduce this yet. What completion UI do you use?

**Comment by oantolin** [comment]
Ah, got it, if you using a quitting action and have a very small value of `echo-keystrokes` (below 0.2 in my tests, but this probably depends on your computer, completion UI and Emacs version) then this happens. Curious.

**Comment by codesuki** [comment]
Thanks for investigating! And sorry for giving too little context. I am using Vertico, Corfu. Emacs is version 29.0.50, would have to check the actual commit :)

It's fine because it actually doesn't do anything. It doesn't even say `RET-s is undefined` which it normally should say.  I could probably increase `echo-keystrokes`, it's 0.25 right now. 

But, I agree it's curious. I don't know about the implementation that drives keystroke echoing, but I imagine it should react properly on the quit signal / command finishing. Somehow Emacs seems to lose track.

**Comment by oantolin** [comment]
I just want to confirm that as you suspected this *is* caused by using `set--this-command-keys` in `embark--act`. That was done to fix #182. It would be nice to figure out a different fix for #182 that didn't have this lingering keys issue.

**Comment by codesuki** [comment]
Oh, that's kinda good news :) 
I tried to reproduce the issue in #182, but my `browse-url` doesn't work even without embark. 😅

**Comment by montag451** [comment]
Hi @oantolin,

Do you have a plan to fix this issue ?

**Comment by oantolin** [comment]
I don't know _how_ to fix it and am open to suggestions.

**Comment by montag451** [comment]
I can have a look but to be honest if you didn't find a solution to fix this issue I don't think I will be able to find one :). Anyway it's not a big issue so it's not a big deal if it's not fixed

**Comment by ywwry66** [comment]
The current fix to #182 is not ideal as `set--this-command-keys` is clearly marked as "Internal use only". And the MWE in #182 does not even work on the current master due to a different error: `embark PCH: (user-error "Minibuffer window is not active")`. So it might be worthwhile to revert bc405f0 before a proper fix to #182 (which could be quite complicated) is proposed.

**Comment by abgruszecki** [comment]
I've hit this problem too. It made me suspect that there's a bug in embark and it leaves unfinished keystroke on the "input stack" (or whatever), which made me use it less because I thought I need to do extra keypresses to "disarm" the leftover keystrokes. It'd be great if the issue could be fixed.

Here's how I reproduced it:
1. On a fresh Emacs install, install `vertico` and `embark` 
2. Bind `C-.` to `embark-act`
3. `(setopt echo-keystrokes 0.02)`
4. Visit a non-existent file
5. `M-x display-buffer`, pick the newly created file buffer
6. `k` – kill the buffer
7.  `y` – confirm (this is b/c of `embark--confirm` I believe?)
8. The minibuffer will now display `y-`

**Comment by LemonBreezes** [comment]
Yes, I agree this needs to be fixed because Doom Emacs has the `echo-keystrokes` set to 0.02, which is a good UI choice to me, but causes this issue to occur for thousands of people.

**Comment by oantolin** [comment]
As I've said before, I am clueless on how to fix this and would appreciate any help I can get in the matter. Also, I'd like to stress this issue is maybe a little annoying but completely inconsequential: you can just ignore the keystrokes echoed in the echo area and go on with your life, no keystrokes get swallowed or misinterpreted.

**Comment by oantolin** [comment]
This is a frustrating problem: I don't like the incorrectly echoed keys either, and I've tried a bunch of things, but nothing seem to work without reintroducing the other bug.

**Comment by LemonBreezes** [comment]
@codesuki Hello. Since we already merged a fix, can you please close this issue?
